python/shotgun_fields/shotgun_field_editable.py

```diff
@@ -0,0 +1,320 @@
+        :type display_widget: :class:`~PySide.QtGui.QWidget`
+        :param parent: The parent widget or ``None``
+        :type parent: :class:`~PySide.QtGui.QWidget`
+        """
+
+        super(ShotgunFieldNotEditable, self).__init__(parent)
+
+        self._display_widget = display_widget
+
+        # this is the "no edit" label that will show on hover
+        self._no_edit_lbl = QtGui.QLabel(self)
+        self._no_edit_lbl.setPixmap(
+            QtGui.QPixmap(":/qtwidgets-shotgun-fields/not_editable.png"))
+        self._no_edit_lbl.setFixedSize(QtCore.QSize(16, 16))
+        self._no_edit_lbl.hide()
+
+        spacer = QtGui.QWidget()
+        spacer.setFixedHeight(self._no_edit_lbl.height())
+        spacer.setFixedWidth(4)
+
+        layout = QtGui.QHBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.setSpacing(0)
+        layout.addWidget(display_widget)
+        layout.addWidget(spacer)
+        layout.addWidget(self._no_edit_lbl)
+        layout.addStretch(10)
+
+        self.installEventFilter(self)
+
+    def eventFilter(self, obj, event):
+        """
+        Filter mouse enter/leave events in order to show/hide the "no edit" label.
+        """
+        if event.type() == QtCore.QEvent.Enter:
+            self._no_edit_lbl.show()
+        elif event.type() == QtCore.QEvent.Leave:
+            self._no_edit_lbl.hide()
+
+        return False
+
+
+class _DisplayWidget(QtGui.QWidget):
+    """
+    A wrapper around a display widget with a hoverable "edit" button.
+    """
+
+    edit_requested = QtCore.Signal()
+
+    def __init__(self, display_widget, parent=None):
+        """
+        Initialize the wrapper widget.
+
+        :param display_widget: The ``DISPLAY`` widget instance
+        :type display_widget: :class:`~PySide.QtGui.QWidget`
+        :param parent: The parent widget instance or None
+        :type parent: :class:`~PySide.QtGui.QWidget`
+        :return:
+        """
+
+        super(_DisplayWidget, self).__init__(parent)
+
+        self._display_widget = display_widget
+
+        self._edit_btn = QtGui.QPushButton()
+        self._edit_btn.setIcon(QtGui.QIcon(":/qtwidgets-shotgun-fields/edit_field.png"))
+        self._edit_btn.setFixedSize(QtCore.QSize(16, 16))
+        self._edit_btn.setFocusPolicy(QtCore.Qt.NoFocus)
+        self._edit_btn.hide()
+
+        # make sure there's never a bg color or border
+        self._edit_btn.setStyleSheet("background-color: none; border: none;")
+
+        spacer = QtGui.QWidget()
+        spacer.setFixedHeight(self._edit_btn.height())
+        spacer.setFixedWidth(4)
+
+        layout = QtGui.QHBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.setSpacing(0)
+        layout.addWidget(display_widget)
+        layout.addWidget(spacer)
+        layout.addWidget(self._edit_btn)
+        layout.addStretch(10)
+
+        self.setMinimumHeight(self._edit_btn.height())
```

Comment:
Because the button only shows on hover, this is required. Otherwise things get jumpy.